packages/discovery-provider/integration_tests/notifications/test_manager_mode_notifs.py

```diff
@@ -0,0 +1,101 @@
+    entities = {
+        "developer_apps": [
+            {
+                "user_id": 5,
+                "name": "My App",
+                "address": "0x3a388671bb4D6E1Ea08D79Ee191b40FB45A8F4C4",
+            },
+            {
+                "user_id": 4,
+                "name": "My App",
+                "address": "0x04c9fc3784120f50932436f84c59aebebb12e0d",
+            },
+        ],
+        "grants": [
+            # Grant created from user to developer app. Should NOT trigger notif:
+            {
+                "user_id": 6,
+                "grantee_address": "0x04c9fc3784120f50932436f84c59aebebb12e0d",
+            },
+            # Grant created from user to user:
+            # This should trigger a notification to the requested manager:
+            {
+                "user_id": 2,
+                "grantee_address": "user1wallet",
+                "created_at": now,
+                "updated_at": now,
+            },
+            # Manager approves request, should trigger a notification to the managed account
+            {
+                "user_id": 2,
+                "grantee_address": "user1wallet",
+                "is_approved": True,
+                "created_at": now,
+                "updated_at": now + timedelta(days=1),
+            },
+            # Manager relationship is removed - no notification
+            {
+                "user_id": 2,
+                "grantee_address": "user1wallet",
+                "is_approved": True,
+                "created_at": now,
+                "updated_at": now + timedelta(days=2),
+                "is_revoked": True,
+            },
+            # Managed account re-requests the manager - should trigger a notification to the manager
+            {
+                "user_id": 2,
+                "grantee_address": "user1wallet",
+                "created_at": now + timedelta(days=3),
+                "updated_at": now + timedelta(days=3),
+                "is_revoked": False,
+            },
+            # Manager approves new request - should trigger a notification to the managed account
+            {
+                "user_id": 2,
+                "grantee_address": "user1wallet",
+                "is_approved": True,
+                "created_at": now + timedelta(days=3),
+                "updated_at": now + timedelta(days=4),
+                "is_revoked": False,
+            },
+        ],
+    }
+    populate_mock_db(db, entities)
+
+    with db.scoped_session() as session:
+        manager_notification: List[Notification] = (
+            session.query(Notification)
+            .filter(Notification.type == "request_manager")
+            .all()
+        )
+        assert len(manager_notification) == 2
+        managed_account_notification: List[Notification] = (
+            session.query(Notification)
+            .filter(Notification.type == "approve_manager_request")
+            .all()
+        )
+        assert len(managed_account_notification) == 2
```

Comment:
nit: Can we check the contents of the notifications as well? This feels like it's not quite checking the "no triggers for specific cases" requirements above.

Reply:
yess good call